langserver/langserver_test.go

```diff
@@ -1428,13 +1426,15 @@ func lspTests(t testing.TB, ctx context.Context, h *LangHandler, c *jsonrpc2.Con
 				hoverTest(t, ctx, c, util.PathToURI(tmpRootPath), pos, want)
 			})
 		}
+
+		h.config.UseBinaryPkgCache = false
+
+		// TODO(anjmao): autocomplete tests fails if binary cache is used
```

Comment:
As gocode is using binary cache for completion if I set this to UseBinaryPkgCache = true a lot of  completion tests are failing since autocomplete returns empty result because test case binary is not built yet at that moment. Is there any way I can pre build each test case code during mountFS?

Reply:
The tests already install the packages:

https://github.com/sourcegraph/go-langserver/blob/68907e07ef2b8479a592485f63a225c36e23ed67/langserver/langserver_test.go#L1401-L1409

Maybe there is another problem that I cannot see at the moment...

Reply:
If I understand correctly, this is only failing because `$GOPATH/pkg` would be used when `UseBinaryPkgCache = true`, and in the case of the tests here we are not actually building the code with `go install` because the code is not on disk. i.e. this wouldn't actually fail in practice / in real use when `UseBinaryPkgCache = true`, yeah?

If so, I think it is fine to not address this for now.

Reply:
Yeah, I am wrong here (I submitted my reply before I saw @lloiser's reply). We are copying the test sources into a temporary GOPATH and `go install`'ing them, so it should be working

Reply:
@lloiser Thanks, yes it's actually building code and pkg binary cache files are created during tests, but for some reasons imported packages scope is empty. 

In gocode package imports prints empty scope if binary cache is used, but it works perfectly fine in vscode when using binary cache.
```
pkg.Imports()[0].Scope()
```
I'm currently investigating `go/importer` used in gocode.